### ISSUE TITLE
20.1 Exception when aggregating a column that does not exist in every part

### ISSUE THREAD

**Issue by SaltTan** [ClickHouse/ClickHouse#9132]
20.1.4.14
The 'rcount' column was added to the table in November:
```
select partition, name from system.parts_columns where column like 'rcount'
partition | name          
----------+---------------
201911    | 201911_10_10_0
201912    | 201912_4_4_0  
202002    | 202002_0_0_0  
202002    | 202002_1_1_0  
```
When CH needs to process older and newer parts together I get an exception or a segfault.
A segfault example:
```
2020.02.14 18:31:19.455661 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Debug> executeQuery: (from [::ffff:10.253.59.13]:58671, user: default) SELECT sum(rcount)  FROM dsp_fact_event_shard PREWHERE (account_id IN (2276648,-1)) WHERE access_day >= '2019-02-02' FORMAT TabSeparatedWithNamesAndTypes;
2020.02.14 18:31:19.456758 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Debug> dw.dsp_fact_event_shard (SelectExecutor): Key condition: unknown, unknown, and, unknown, and
2020.02.14 18:31:19.456775 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Debug> dw.dsp_fact_event_shard (SelectExecutor): MinMax index condition: unknown, (column 0 in [17929, +inf)), and, unknown, and
2020.02.14 18:31:19.456785 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Debug> dw.dsp_fact_event_shard (SelectExecutor): Selected 11 parts by date, 11 parts by key, 231 marks to read from 11 ranges
2020.02.14 18:31:19.457860 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> dw.dsp_fact_event_shard (SelectExecutor): Reading approx. 1892352 rows with 4 streams
2020.02.14 18:31:19.458055 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
2020.02.14 18:31:19.458146 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Debug> executeQuery: Query pipeline:
Expression
 Expression
  ParallelAggregating
   Expression <C3><97> 4
    Filter
     MergeTreeThread

2020.02.14 18:31:19.458182 [ 56 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> ParallelAggregatingBlockInputStream: Aggregating
2020.02.14 18:31:19.458726 [ 54 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> Aggregator: Aggregation method: without_key
2020.02.14 18:31:19.458752 [ 47 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> Aggregator: Aggregation method: without_key
2020.02.14 18:31:19.458774 [ 44 ] {a3f4a1cb-e769-4e2f-abeb-258e08b7556a} <Trace> Aggregator: Aggregation method: without_key
2020.02.14 18:31:19.458935 [ 58 ] {} <Fatal> BaseDaemon: ########################################
2020.02.14 18:31:19.458978 [ 58 ] {} <Fatal> BaseDaemon: (version 20.1.4.14 (official build)) (from thread 23) (query_id: a3f4a1cb-e769-4e2f-abeb-258e08b7556a) Received signal Segmentation fault (11).
2020.02.14 18:31:19.458996 [ 58 ] {} <Fatal> BaseDaemon: Address: 0x18 Access: read. Address not mapped to object.
2020.02.14 18:31:19.459013 [ 58 ] {} <Fatal> BaseDaemon: Stack trace: 0x8e549ae 0x97a6605 0x97a9d7f 0x976c2d6 0x976dad3 0x91a7ef7 0x91da0f4 0x91da313 0x8a6f42f 0x94a9a11 0x8a6f42f 0x949f368 0x8a6f42f 0x94ec720 0x94ecca5 0x4fa4fd7 0x4fa358f 0x7f5e7e871494 0x7f5e7e1a7acf
2020.02.14 18:31:19.459067 [ 58 ] {} <Fatal> BaseDaemon: 3. 0x8e549ae DB::ColumnVector<int>::filter(DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, long) const  in /usr/bin/clickhouse
2020.02.14 18:31:19.459089 [ 58 ] {} <Fatal> BaseDaemon: 4. 0x97a6605 DB::MergeTreeRangeReader::filterColumns(std::__1::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn> > >&, DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) const  in /usr/bin/clickhouse
2020.02.14 18:31:19.459103 [ 58 ] {} <Fatal> BaseDaemon: 5. 0x97a9d7f DB::MergeTreeRangeReader::read(unsigned long, std::__1::vector<DB::MarkRange, std::__1::allocator<DB::MarkRange> >&)  in /usr/bin/clickhouse
2020.02.14 18:31:19.459118 [ 58 ] {} <Fatal> BaseDaemon: 6. 0x976c2d6 DB::MergeTreeBaseSelectProcessor::readFromPartImpl()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459130 [ 58 ] {} <Fatal> BaseDaemon: 7. 0x976dad3 DB::MergeTreeBaseSelectProcessor::generate()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459142 [ 58 ] {} <Fatal> BaseDaemon: 8. 0x91a7ef7 DB::ISource::work()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459154 [ 58 ] {} <Fatal> BaseDaemon: 9. 0x91da0f4 DB::TreeExecutorBlockInputStream::execute()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459172 [ 58 ] {} <Fatal> BaseDaemon: 10. 0x91da313 DB::TreeExecutorBlockInputStream::readImpl()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459186 [ 58 ] {} <Fatal> BaseDaemon: 11. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459197 [ 58 ] {} <Fatal> BaseDaemon: 12. 0x94a9a11 DB::FilterBlockInputStream::readImpl()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459208 [ 58 ] {} <Fatal> BaseDaemon: 13. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459219 [ 58 ] {} <Fatal> BaseDaemon: 14. 0x949f368 DB::ExpressionBlockInputStream::readImpl()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459229 [ 58 ] {} <Fatal> BaseDaemon: 15. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
2020.02.14 18:31:19.459238 [ 58 ] {} <Fatal> BaseDaemon: 16. 0x94ec720 DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::thread(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long)  in /usr/bin/clickhouse
2020.02.14 18:31:19.459257 [ 58 ] {} <Fatal> BaseDaemon: 17. 0x94ecca5 ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*, std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long&>(void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&, unsigned long&)::'lambda'()::operator()() const  in /usr/bin/clickhouse
2020.02.14 18:31:19.459275 [ 58 ] {} <Fatal> BaseDaemon: 18. 0x4fa4fd7 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>)  in /usr/bin/clickhouse
2020.02.14 18:31:19.459282 [ 58 ] {} <Fatal> BaseDaemon: 19. 0x4fa358f ?  in /usr/bin/clickhouse
2020.02.14 18:31:19.459295 [ 58 ] {} <Fatal> BaseDaemon: 20. 0x7494 start_thread  in /lib/x86_64-linux-gnu/libpthread-2.24.so
2020.02.14 18:31:19.459306 [ 58 ] {} <Fatal> BaseDaemon: 21. 0xe8acf clone  in /lib/x86_64-linux-gnu/libc-2.24.so
```
An exception example:
```
2020.02.14 18:30:54.284219 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> executeQuery: (from [::ffff:10.253.59.13]:58650, user: default) SELECT sum(rcount)  FROM dsp_fact_event_shard PREWHERE (account_id IN (2276648,-1)) WHERE access_day = '2019-02-02' FORMAT TabSeparatedWithNamesAndTypes;
2020.02.14 18:30:54.285674 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> dw.dsp_fact_event_shard (SelectExecutor): Key condition: unknown, unknown, and, unknown, and
2020.02.14 18:30:54.285693 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> dw.dsp_fact_event_shard (SelectExecutor): MinMax index condition: unknown, (column 0 in [17929, 17929]), and, unknown, and
2020.02.14 18:30:54.285702 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> dw.dsp_fact_event_shard (SelectExecutor): Selected 1 parts by date, 1 parts by key, 208 marks to read from 1 ranges
2020.02.14 18:30:54.285918 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Trace> dw.dsp_fact_event_shard (SelectExecutor): Reading approx. 1703936 rows with 4 streams
2020.02.14 18:30:54.286104 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
2020.02.14 18:30:54.286196 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> executeQuery: Query pipeline:
Expression
 Expression
  ParallelAggregating
   Expression <C3><97> 4
    Filter
     MergeTreeThread

2020.02.14 18:30:54.286232 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Trace> ParallelAggregatingBlockInputStream: Aggregating
2020.02.14 18:30:54.286928 [ 44 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Trace> Aggregator: Aggregation method: without_key
2020.02.14 18:30:54.287829 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Error> executeQuery: Code: 9, e.displayText() = DB::Exception: Size of filter doesn't match size of column. (version 20.1.4.14 (official build)) (from [::ffff:10.253.59.13]:58650) (in query: SELECT sum(rcount)  FROM dsp_fact_event_shard PREWHERE (account_id IN (2276648,-1)) WHERE access_day = '2019-02-02' FORMAT TabSeparatedWithNamesAndTypes;), Stack trace (when copying this message, always include the lines below):

0. 0xbc34d6c Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
1. 0x4f6e129 DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
2. 0x8e54f4e DB::ColumnVector<int>::filter(DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, long) const  in /usr/bin/clickhouse
3. 0x97a6605 DB::MergeTreeRangeReader::filterColumns(std::__1::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn> > >&, DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) const  in /usr/bin/clickhouse
4. 0x97a9d7f DB::MergeTreeRangeReader::read(unsigned long, std::__1::vector<DB::MarkRange, std::__1::allocator<DB::MarkRange> >&)  in /usr/bin/clickhouse
5. 0x976c2d6 DB::MergeTreeBaseSelectProcessor::readFromPartImpl()  in /usr/bin/clickhouse
6. 0x976dad3 DB::MergeTreeBaseSelectProcessor::generate()  in /usr/bin/clickhouse
7. 0x91a7ef7 DB::ISource::work()  in /usr/bin/clickhouse
8. 0x91da0f4 DB::TreeExecutorBlockInputStream::execute()  in /usr/bin/clickhouse
9. 0x91da313 DB::TreeExecutorBlockInputStream::readImpl()  in /usr/bin/clickhouse
10. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
11. 0x94a9a11 DB::FilterBlockInputStream::readImpl()  in /usr/bin/clickhouse
12. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
13. 0x949f368 DB::ExpressionBlockInputStream::readImpl()  in /usr/bin/clickhouse
14. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
15. 0x94ec720 DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::thread(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long)  in /usr/bin/clickhouse
16. 0x94ecca5 ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*, std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long&>(void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&, unsigned long&)::'lambda'()::operator()() const  in /usr/bin/clickhouse
17. 0x4fa4fd7 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>)  in /usr/bin/clickhouse
18. 0x4fa358f ?  in /usr/bin/clickhouse
19. 0x7494 start_thread  in /lib/x86_64-linux-gnu/libpthread-2.24.so
20. 0xe8acf clone  in /lib/x86_64-linux-gnu/libc-2.24.so

2020.02.14 18:30:54.287965 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> MemoryTracker: Peak memory usage (total): 649.07 KiB.
2020.02.14 18:30:54.287992 [ 56 ] {97c6b0d3-c459-41f6-b7bd-f75caaa743bc} <Debug> MemoryTracker: Peak memory usage (for query): 4.64 MiB.
2020.02.14 18:30:54.288018 [ 56 ] {} <Error> HTTPHandler: Code: 9, e.displayText() = DB::Exception: Size of filter doesn't match size of column., Stack trace (when copying this message, always include the lines below):

0. 0xbc34d6c Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
1. 0x4f6e129 DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
2. 0x8e54f4e DB::ColumnVector<int>::filter(DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, long) const  in /usr/bin/clickhouse
3. 0x97a6605 DB::MergeTreeRangeReader::filterColumns(std::__1::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn> > >&, DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) const  in /usr/bin/clickhouse
4. 0x97a9d7f DB::MergeTreeRangeReader::read(unsigned long, std::__1::vector<DB::MarkRange, std::__1::allocator<DB::MarkRange> >&)  in /usr/bin/clickhouse
5. 0x976c2d6 DB::MergeTreeBaseSelectProcessor::readFromPartImpl()  in /usr/bin/clickhouse
6. 0x976dad3 DB::MergeTreeBaseSelectProcessor::generate()  in /usr/bin/clickhouse
7. 0x91a7ef7 DB::ISource::work()  in /usr/bin/clickhouse
8. 0x91da0f4 DB::TreeExecutorBlockInputStream::execute()  in /usr/bin/clickhouse
9. 0x91da313 DB::TreeExecutorBlockInputStream::readImpl()  in /usr/bin/clickhouse
10. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
11. 0x94a9a11 DB::FilterBlockInputStream::readImpl()  in /usr/bin/clickhouse
12. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
13. 0x949f368 DB::ExpressionBlockInputStream::readImpl()  in /usr/bin/clickhouse
14. 0x8a6f42f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
15. 0x94ec720 DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::thread(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long)  in /usr/bin/clickhouse
16. 0x94ecca5 ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*, std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long&>(void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&, unsigned long&)::'lambda'()::operator()() const  in /usr/bin/clickhouse
17. 0x4fa4fd7 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>)  in /usr/bin/clickhouse
18. 0x4fa358f ?  in /usr/bin/clickhouse
19. 0x7494 start_thread  in /lib/x86_64-linux-gnu/libpthread-2.24.so
20. 0xe8acf clone  in /lib/x86_64-linux-gnu/libc-2.24.so
 (version 20.1.4.14 (official build))
```
There is no problem with this in 19.16.

**Comment by CurtizJ** [comment]
Same as #9064. Fixed in master

**Comment by SaltTan** [comment]
The segfault is fixed in 20.1.6.30, but the exception is still there:
```
2020.03.10 14:46:27.997915 [ 59 ] {} <Error> HTTPHandler: Code: 9, e.displayText() = DB::Exception: Size of filter doesn't match size of column., Stack trace (when copying this message, always include the lines below):

0. 0x100ac1bc Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
1. 0x8e74849 DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int)  in /usr/bin/clickhouse
2. 0xd35391e DB::ColumnVector<int>::filter(DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&, long) const  in /usr/bin/clickhouse
3. 0xd72ec75 DB::MergeTreeRangeReader::filterColumns(std::__1::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn> > >&, DB::PODArray<unsigned char, 4096ul, Allocator<false, false>, 15ul, 16ul> const&) const  in /usr/bin/clickhouse
4. 0xd7324ff DB::MergeTreeRangeReader::read(unsigned long, std::__1::deque<DB::MarkRange, std::__1::allocator<DB::MarkRange> >&)  in /usr/bin/clickhouse
5. 0xd7289c9 DB::MergeTreeBaseSelectProcessor::readFromPartImpl()  in /usr/bin/clickhouse
6. 0xd72a243 DB::MergeTreeBaseSelectProcessor::generate()  in /usr/bin/clickhouse
7. 0xd8688b7 DB::ISource::work()  in /usr/bin/clickhouse
8. 0xdb59250 DB::SourceWithProgress::work()  in /usr/bin/clickhouse
9. 0xd8b0f94 DB::TreeExecutorBlockInputStream::execute()  in /usr/bin/clickhouse
10. 0xd8b11d3 DB::TreeExecutorBlockInputStream::readImpl()  in /usr/bin/clickhouse
11. 0xcc1e47f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
12. 0xd029951 DB::FilterBlockInputStream::readImpl()  in /usr/bin/clickhouse
13. 0xcc1e47f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
14. 0xd025ac8 DB::ExpressionBlockInputStream::readImpl()  in /usr/bin/clickhouse
15. 0xcc1e47f DB::IBlockInputStream::read()  in /usr/bin/clickhouse
16. 0xd0aa830 DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::thread(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long)  in /usr/bin/clickhouse
17. 0xd0aadb5 ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*, std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long&>(void (DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>::*&&)(std::__1::shared_ptr<DB::ThreadGroupStatus>, unsigned long), DB::ParallelInputsProcessor<DB::ParallelAggregatingBlockInputStream::Handler>*&&, std::__1::shared_ptr<DB::ThreadGroupStatus>&&, unsigned long&)::'lambda'()::operator()() const  in /usr/bin/clickhouse
18. 0x8e97347 ThreadPoolImpl<std::__1::thread>::worker(std::__1::__list_iterator<std::__1::thread, void*>)  in /usr/bin/clickhouse
19. 0x8e9580f ?  in /usr/bin/clickhouse
20. 0x7494 start_thread  in /lib/x86_64-linux-gnu/libpthread-2.24.so
21. 0xe8acf clone  in /lib/x86_64-linux-gnu/libc-2.24.so
 (version 20.1.6.30 (official build))
```

**Comment by SaltTan** [comment]
@CurtizJ I can provide the parts if needed

**Comment by CurtizJ** [comment]
> @CurtizJ I can provide the parts if needed

Yes, it will be helpful.

**Comment by SaltTan** [comment]
@CurtizJ Here are the parts
[i9132.tar.gz](https://github.com/ClickHouse/ClickHouse/files/4320828/i9132.tar.gz)
```
ATTACH TABLE test.i9132 
(
  `access_time`   DateTime,
  `access_day`    Date DEFAULT toDate (access_time),
  `time_key`      Int32,
  `seller_key`    Int32,
  `cmp_key`       Int32,
  `rcount`        Int32 DEFAULT toInt32 (1)
)
ENGINE = MergeTree () PARTITION BY toYYYYMM (access_day)
ORDER BY (time_key,seller_key,cmp_key) SETTINGS index_granularity_bytes = 0;

SELECT sum(rcount)
FROM test.i9132
PREWHERE seller_key IN (2350462);

┌─sum(rcount)─┐
│      955714 │
└─────────────┘

1 rows in set. Elapsed: 0.003 sec. Processed 955.73 thousand rows, 5.54 MB (316.78 million rows/s., 1.84 GB/s.)

SELECT sum(rcount)
FROM test.i9132
PREWHERE seller_key IN (-1);

Received exception from server (version 20.1.6):
Code: 9. DB::Exception: Received from localhost:9000. DB::Exception: Size of filter doesn't match size of column..
```
Sometimes I also get another exception, but I couldn't reproduce it on these parts:
DB::Exception: Invalid number of rows in Chunk column Int32: expected 13, got 65536

**Comment by CurtizJ** [comment]
> Sometimes I also get another exception, but I couldn't reproduce it on these parts:
> DB::Exception: Invalid number of rows in Chunk column Int32: expected 13, got 65536
#9580

**Comment by CurtizJ** [comment]
Also fixed #9580.
